From b00f50b861fac5b04acf2e90af96bd3839c93bbc Mon Sep 17 00:00:00 2001
From: Tobias Schramm <t.schramm@manjaro.org>
Date: Thu, 13 May 2021 16:14:06 +0200
Subject: [PATCH 15/20] backlight: pwm_bl: always parse
 default-brightness-level property

It is often desireable to set a default brightness for backlights as
early as possible. At the moment in device trees this is only possible
when also specifying a brightness table, too.
This commit changes the behaviour of the fw node parsing to always
consider the default-brightness-level property and apply it, even
when there is no brightness table.

Signed-off-by: Tobias Schramm <t.schramm@manjaro.org>
---
 drivers/video/backlight/pwm_bl.c | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/drivers/video/backlight/pwm_bl.c b/drivers/video/backlight/pwm_bl.c
index dfc760830eb9..9381b8d575e1 100644
--- a/drivers/video/backlight/pwm_bl.c
+++ b/drivers/video/backlight/pwm_bl.c
@@ -616,6 +616,18 @@ static int pwm_backlight_probe(struct platform_device *pdev)
 		goto err_alloc;
 	}
 
+	if (device_property_present(&pdev->dev, "default-brightness-level")) {
+		uint32_t value;
+
+		ret = device_property_read_u32(&pdev->dev,
+					       "default-brightness-level",
+					       &value);
+		if (ret < 0)
+			goto err_alloc;
+
+		data->dft_brightness = value;
+	}
+
 	if (data->dft_brightness > data->max_brightness) {
 		dev_warn(&pdev->dev,
 			 "invalid default brightness level: %u, using %u\n",
-- 
2.31.1

